build_tools/fetch_artifacts.py

```diff
@@ -6,20 +6,24 @@
 # but those artifacts may not have all required dependencies.
 
 import argparse
+import concurrent.futures
 import platform
 import subprocess
 import sys
 
 GENERIC_VARIANT = "generic"
 PLATFORM = platform.system()
 
-
+# TODO (geomin12): switch out logging library
 def log(*args, **kwargs):
     print(*args, **kwargs)
     sys.stdout.flush()
 
 
 def s3_bucket_exists(run_id):
+    """
+    This function checks that the AWS S3 bucket exists
+    """
```

Comment:
No need for "This function...", just lead with "Checks that..."

Same for all docstrings.

* https://peps.python.org/pep-0257/
* https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

  * We typically follow descriptive style:
  
    > The docstring may be descriptive-style (`"""Fetches rows from a Bigtable."""`) or imperative-style (`"""Fetch rows from a Bigtable."""`), but the style should be consistent within a file. 

  * Also: 

    > A docstring should be organized as a summary line (one physical line not exceeding 80 characters) terminated by a period, question mark, or exclamation point.

Reply:
Still need to address

> A docstring should be organized as a summary line (one physical line not exceeding 80 characters) terminated by a period, question mark, or exclamation point.

For all docstrings.